assignments/callbacks.js

```diff
@@ -27,29 +27,80 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
 
 function getLength(arr, cb) {
   // getLength passes the length of the array into the callback.
+  return cb(arr.length);
 }
 
+getLength(items, function(len){
+  console.log(len)
+});
+
 function last(arr, cb) {
   // last passes the last item of the array into the callback.
+  return cb(arr[arr.length-1])
 }
 
+last(items, function(last){
+  console.log(last)
+});
+
 function sumNums(x, y, cb) {
   // sumNums adds two numbers (x, y) and passes the result to the callback.
+  sum = x+y;
+  return cb(sum);
 }
 
+sumNums(2, 3, function(s){
+  console.log(s) //could also just console.log sum as the variable defined in sumNums and not pass a parameter
+});
+
 function multiplyNums(x, y, cb) {
   // multiplyNums multiplies two numbers and passes the result to the callback.
+  multi = x*y;
+  return cb(multi);
 }
 
+multiplyNums(2, 3, function(product){
+  console.log(product)
+});
+
 function contains(item, list, cb) {
   // contains checks if an item is present inside of the given array/list.
   // Pass true to the callback if it is, otherwise pass false.
+ if(list.includes(item)){
+   return cb(true);
+ }
+ else{
+   return cb(false);
+ }
 }
 
+//true check
+contains('Notebook', items, function(item) {
+  console.log(item);
+});
+
+//false check
+contains('Poop', items, function(item) {
+  console.log(item);
+});
+
 /* STRETCH PROBLEM */
 
 function removeDuplicates(array, cb) {
   // removeDuplicates removes all duplicate values from the given array.
   // Pass the duplicate free array to the callback function.
   // Do not mutate the original array.
+  
+sortedArr = array.sort(); //sort array in order
+for (let i=0; i<sortedArr.length; i++){ //loop through each element
+  if(sortedArr[i]=== sortedArr[i+1]){ //if index is the same as the next index 
+    sortedArr.splice(i, 1); //splice that index out 
+  }
+}
+return cb(sortedArr);
+}
+
```

Comment:
very interesting way to remove duplicates.  great effort!  however, this code only accounts for if the values are the same for elements adjacent to one another.  what if the values are the same at the beginning and the end of the array?  your code doesn't account for that case.  

you can filter out items where the indexOf(arr[i]) === i.  this condition will look for the first index position where the element occurs, and if that === the current index position, then that needs to be kept because it is unique to the array.  a 2nd occurence of the letter will not make it equal to where you are with the current index, so it means it's not unique.

Reply:
![image](https://user-images.githubusercontent.com/30554629/59484707-87d97900-8e38-11e9-98a9-7d2c8681871e.png)